server/src/cli/permission_groups.rs

```diff
@@ -0,0 +1,106 @@
+use cli_table::{Cell as _, Style as _, Table as _};
+use cosmicverge_shared::permissions::{Permission, Service};
+use database::schema::{PermissionGroup, Statement};
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(about = "commands to execute")]
+/// commands that the server can execute
+pub struct Command {
+    /// The name of the permission group
+    group_name: String,
+
+    #[structopt(subcommand)]
+    operation: Operation,
+}
+
+#[derive(StructOpt, Debug)]
+enum Operation {
+    /// Creates a new permission group
+    Create,
+    /// Views a permission group
+    View,
+    /// Adds a permission
+    Add { permission: Permission },
+    /// Removes a permission
+    Remove { permission: Permission },
+    /// Grants all permissions to a service
+    AddService { service: Service },
+    /// Removes all permissions from a service
+    RemoveService { service: Service },
+}
+
+pub async fn handle_command(group_command: Command) -> anyhow::Result<()> {
```

Comment:
On that note, I belive this sould be a method too, what do you think?
Personally I prefer functions to not lie around if they can only be used together with a specific `struct` or `trait`, then they should be part of them, even as associated methods.

Reply:
I just think of how function is being called and the pattern I'm trying to achieve. Can this be done with traits? Yes, but it's just extra lines of code. In the end, the consumer will be doing:

````
PermissionGroups::handle_command(command).await
or
permission_groups::handle_command(command).await
```

which to me don't really "feel" different. Additionally, I don't love the idea of another type named PermissionGroups clogging up our Search by Symbol in our Editors, but I'm ok with a namespaced permission_groups mod.

Reply:
I meant (incl. questionable bikeshedding):
```rust
command.handle()
```
But I can see why you would like that command to be in `PermissionGroups`, I agree in that case though, that wouldn't make any sense.